verifier/src/main/java/com/aws/jverify/verifier/NameMangler.java

```diff
@@ -0,0 +1,113 @@
+package com.aws.jverify.verifier;
+
+import com.sun.source.tree.Tree;
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.code.Type.*;
+
+import com.sun.tools.javac.util.Names;
+import com.sun.tools.javac.tree.JCTree;
+
+import java.util.HashSet;
+
+/**
+ * Update the name of all method and field names of symbols within a compilation unit
+ * to ensure uniqueness. The mangling algorithm is deterministic and works as follows:
+ *   - a prefix ("_F") is added to a field name
+ *   - a prefix ("_Z") is added to method names (except constructors)
+ *   - a suffix is added to all method names. This suffix encodes the parameter types, as given by the typeMangling
+ *     method
+ */
+public class NameMangler {
+    private final JavaToDafnyCompiler javaToDafnyCompiler;
+    private Names nameFactory;
+    static private final String fieldPrefix = "F_";
+    static private final String methodPrefix = "Z_";
+
+    // Cache of all symbols whose names were already mangled.
+    private HashSet<Symbol> mangledSymbols;
```

Comment:
```suggestion
    private final Set<Symbol> mangledSymbols;
```


(More Java idiomatic to use the general collections interfaces like `Set` rather than specific implementations when that detail doesn't matter, which is almost all the time)